pkg/destroy/gcp/address.go

```diff
@@ -84,19 +115,25 @@ func (o *ClusterUninstaller) deleteAddress(ctx context.Context, item cloudResour
 // destroyAddresses removes all address resources that have a name prefixed
 // with the cluster's infra ID.
 func (o *ClusterUninstaller) destroyAddresses(ctx context.Context) error {
-	found, err := o.listAddresses(ctx)
-	if err != nil {
-		return err
-	}
-	items := o.insertPendingItems("address", found)
-	for _, item := range items {
-		err := o.deleteAddress(ctx, item)
+	for _, scope := range []resourceScope{gcpGlobalResource, gcpRegionalResource} {
+		found, err := o.listAddresses(ctx, scope)
 		if err != nil {
-			o.errorTracker.suppressWarning(item.key, err, o.Logger)
+			return fmt.Errorf("failed to list %s addresses: %w", scope, err)
+		}
+		items := o.insertPendingItems("address", found)
+		for _, item := range items {
+			err := o.deleteAddress(ctx, item, scope)
+			if err != nil {
+				o.errorTracker.suppressWarning(item.key, err, o.Logger)
+			}
+		}
+		if items = o.getPendingItems("address"); len(items) > 0 {
+			for _, item := range items {
+				if err := o.deleteAddress(ctx, item, scope); err != nil {
+					return fmt.Errorf("error deleting pending address %s: %w", item.name, err)
```

Comment:
Instead of returning at the first error, shouldn't you save the errors to return later but keep deleting pending items?

Reply:
This is the same process as the others. This will be investigated in a future PR where the destroy code will be generalized.

Reply:
Fair enough.